### DISCUSSION TITLE
Copy ROMs / NVDIA Card necessary?

### DISCUSSION THREAD

**Discussion by jayphizzle** [games-on-whales/gow#19] [Q&A]
Hi!

I just found your awesome project. I'm just into retro gaming & retroarch and currently searching for a sufficiently performant device which can run most of the games. If i could all run in a vm on my server would be great.

So currently i have to questions:

How can i copy files (roms) to retroarch? (With Batocera i can access the rom folders via samba shares) ?
Is a NVIDIA Card necessary or can i also run GOW on a Intel iGPU (Passthrough via GVT-g)?

Thanks!

**Comment by ABeltramo** [comment]
Hello and welcome! 👋 

I never used [Batocera](https://batocera.org/), it looks very cool! 
We are nowhere near that level of ease of use with a fully batteries included environment (yet? 😄). The main difference I can see is that they provide a fully finished OS image that you have to install on a bare system, where you'll play games locally with a plugged in monitor and keyboard/joypad etc etc. 

We are taking a different approach: a remote server that can stream games (or generally any graphical application) to [all Moonlight supported clients](https://moonlight-stream.org/) that can be: your phone, your laptop, or even a Nintendo Switch! You can pick any OS that supports Docker and start using GOW. You can then install other applications or run any other Docker container you might need as you'll do in a personal server.

> If i could all run in a vm on my server would be great

This is definitely our end goal too.

> How can i copy files (roms) to retroarch? (With Batocera i can access the rom folders via samba shares) ?

You have many ways to put the files there, think about this as a remote desktop environment: you can mount a samba share, you can use `scp` if you have an ssh connection, plug a USB stick on your server, etc. etc. 
All of these is outside of our scope, we just map a folder from the host OS to the sandboxed RetroArch instance running inside Docker ([see `${local_state}` in the docker-compose configuration](https://github.com/games-on-whales/gow/blob/00793be56f145136c1c46781f8f8386dc4c04884/docker-compose.yml#L74))

>  Is a NVIDIA Card necessary or can i also run GOW on a Intel iGPU (Passthrough via GVT-g)?

A GPU isn't strictly necessary, you can run everything even without any GPU at all! 
In that case everything will be [software rendered](https://en.wikipedia.org/wiki/Software_rendering) by your CPU, a great example of that is to browse the internet using Firefox.
Adding a GPU (or iGPU) will give you [hardware acceleration](https://en.wikipedia.org/wiki/Hardware_acceleration) which is really needed by the more advanced games and applications.
We are currently testing different GPU configurations to see what works and what needs to be fixed.


I hope this will clear some doubt, feel free to join our [Discord server](https://discord.gg/kRGUDHNHt2) for a chat if you have any more questions.
